ext/mysql2/client.c

```diff
@@ -880,7 +880,7 @@ static VALUE rb_mysql_client_server_info(VALUE self) {
  *
  * Return the file descriptor number for this client.
  */
-static VALUE rb_mysql_client_socket(VALUE self) {
+static VALUE rb_mysql_client_socket(RB_MYSQL_UNUSED VALUE self) {
```

Comment:
In pg.gem we have
```C
#define UNUSED(x) ((void)(x))
```
and we mark the variable as
```C
UNUSED(self);
```
in the branch, that doesn't use this variable.
This was my first thought, but then I noticed `RB_MYSQL_UNUSED` .